typegate/src/services/auth/protocols/oauth2.ts

```diff
@@ -13,17 +13,84 @@ import {
   setEncryptedSessionCookie,
 } from "../cookies.ts";
 import { Protocol } from "./protocol.ts";
-import { DenoRuntime } from "../../../runtimes/deno/deno.ts";
-import { Auth, Materializer } from "../../../typegraph/types.ts";
+import { Auth } from "../../../typegraph/types.ts";
+import { FunctionNode } from "../../../typegraph/type_node.ts";
+import { ComputeStage } from "../../../engine/query_engine.ts";
+import * as ast from "graphql/ast";
+import {
+  generateValidator,
+  generateWeakValidator,
+} from "../../../engine/typecheck/input.ts";
 
 const logger = getLogger(import.meta);
 
+class AuthProfiler {
+  constructor(
+    private authParameters: AdditionalAuthParams,
+    private funcIndex: number,
+  ) {}
+
+  private getComputeStage(): ComputeStage {
+    const { tg, runtimeReferences } = this.authParameters;
+    const funcNode = tg.type(this.funcIndex) as FunctionNode;
```

Comment:
`tg.type(this.funcIndex, Type.FUNCTION)` would make a validation and the type assertion.